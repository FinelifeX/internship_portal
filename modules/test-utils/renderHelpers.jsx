/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import TR from 'react-test-renderer';
import { render as RTLRender } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

/**
 * @typedef {Object.<string, any>} Props
 */

/**
 * @typedef {React.ReactNode} Children
 */

/**
 * @typedef {{props?: Props, children?: Children}} RenderOptions
 */

/**
 *
 * @param {React.ComponentType} Component
 * @param {RenderOptions} [options]
 * @returns {JSX.Element}
 */
export const render = (Component, options = {}) => {
  const { props, children } = options;
  return <Component {...props}>{children}</Component>;
};

/**
 * Render React component inside MemoryRouter
 * @param {React.ComponentType} Component
 * @param {RenderOptions} [options]
 * @param {import('react-router-dom').MemoryRouterProps} routerOptions
 * @returns {JSX.Element}
 */
export const renderWithRouter = (
  Component,
  options = {},
  routerOptions = {},
) => {
  const { props, children } = options;

  return (
    <MemoryRouter {...routerOptions}>
      <Component {...props}>{children}</Component>
    </MemoryRouter>
  );
};

/**
 * Shallow render component using "react-test-renderer"
 * @type {import('react-test-renderer').create}
 */
export const shallow = TR.create;

/**
 * Render component using "render()" from "@testing-library/react"
 * @type {import('@testing-library/react').render}
 */
export const mount = RTLRender;
