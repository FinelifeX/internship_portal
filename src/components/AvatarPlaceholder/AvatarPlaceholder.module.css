.circle {
  border-radius: 50%;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background-color: #e8e8e8;
}

.letters {
  font-size: 50%;
  color: #011b4a;
}
