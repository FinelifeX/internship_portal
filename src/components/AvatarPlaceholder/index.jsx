import React, { useMemo } from 'react';
import PT from 'prop-types';
import styles from './AvatarPlaceholder.module.css';

const AvatarPlaceholder = ({ fullName, size }) => {
  const letters = useMemo(() => {
    if (!fullName) return null;

    return fullName
      .split(' ')
      .slice(0, 2)
      .reduce((prev, curr) => prev + curr[0], '');
  }, [fullName]);

  return (
    <div className={styles.circle} style={{ width: size, height: size }}>
      <span data-testid="text" className={styles.letters}>
        {letters}
      </span>
    </div>
  );
};

AvatarPlaceholder.defaultProps = {
  fullName: null,
  size: '22px',
};

AvatarPlaceholder.propTypes = {
  fullName: PT.string,
  size: PT.oneOfType([PT.string, PT.number]),
};

export default AvatarPlaceholder;
