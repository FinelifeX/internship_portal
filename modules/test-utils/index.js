import * as RenderHelpers from './renderHelpers';

export { RenderHelpers };
