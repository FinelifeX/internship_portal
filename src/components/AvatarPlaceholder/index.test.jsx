import { shallow, mount, render } from 'test-utils/renderHelpers';
import AvatarPlaceholder from './index';

describe('AvatarPlaceholder component', () => {
  const TEST_ID_TEXT = 'text';

  const defaultProps = {
    fullName: 'Adam Jensen',
  };
  const renderComponent = (props) =>
    render(AvatarPlaceholder, { defaultProps, props });

  it('should render correctly', () => {
    const tree = shallow(renderComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should not contain letters', () => {
    const { getByTestId } = mount(renderComponent({ fullName: null }));

    expect(getByTestId(TEST_ID_TEXT)).toBeEmptyDOMElement();
  });

  it.each([
    { fullName: 'Ross Geller', expected: 'RG' },
    { fullName: 'Monica Geller', expected: 'MG' },
    { fullName: 'Phoebe Buffay', expected: 'PB' },
  ])(
    'should render transformed name correctly for $fullName - $expected',
    ({ fullName, expected }) => {
      const { getByTestId } = mount(renderComponent({ fullName }));

      expect(getByTestId(TEST_ID_TEXT)).toHaveTextContent(expected);
    },
  );
});
