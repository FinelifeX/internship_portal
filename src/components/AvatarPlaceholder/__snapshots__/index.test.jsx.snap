// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarPlaceholder component should render correctly 1`] = `
<div
  className="circle"
  style={
    Object {
      "height": "22px",
      "width": "22px",
    }
  }
>
  <span
    className="letters"
    data-testid="text"
  >
    AJ
  </span>
</div>
`;
